### PR TITLE
Fix wrong viewport getting restored when navigation between screens

### LINKED ISSUE #1859: Wrong map viewport is restored
@shobhitagarwal1612 @JSunde  FYI in case you have ideas

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1859

Use SharedFlow instead of StateFlow for updating camera position whenever survey or current location changes.

@gino-m  PTAL?

### CODE REVIEW

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks for the fix! One design question.

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt [hunk 1]
```diff
@@ -198,8 +198,11 @@ constructor(
     }
   }
 
+  /** Emits a stream of location lock state. */
+  fun getLocationLockStatus(): LiveData<Result<Boolean>> = locationLock.asLiveData()
+
   /** Emits a stream of camera update requests. */
-  fun getCameraUpdateRequests(): Flow<CameraUpdateRequest> = _cameraUpdateRequests.filterNotNull()
+  fun getCameraUpdateRequests(): LiveData<CameraUpdateRequest> = _cameraUpdateRequests
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:205]: So should we try using a SharedFlow instead?

[the remark this answers is not in the document]
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:205]: Probably yes, provided we don't need to replay the last value to callers. If that's the case we should clarify how this event and related state are modeled.

[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt [hunk 2]
```diff
@@ -214,36 +207,34 @@ constructor(
...
+  private fun updateCameraPositionOnLocationChange(): Flow<CameraUpdateRequest> =
     locationLock
       .flatMapLatest { enabled ->
         getLocationUpdates()
           .map { it.toCoordinates() }
           .filter { enabled.getOrDefault(false) }
           .withIndex()
       }
-      .collect { (index, coordinates) ->
+      .map { (index, coordinates) ->
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:218]: Can we extract this lambda into its own function? eg `onLocationUpdate()`?

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt]: Done

**Review by gino-m — APPROVED** [review]

**Pull request merged**